pkg/kubelet/util/pluginwatcher/README

```diff
@@ -0,0 +1,29 @@
+This folder contains a utility, pluginwatcher, for Kubelet to register
+different types of node-level plugins such as device plugins or CSI plugins.
+It discovers plugins by monitoring inotify events under the directory returned by
+kubelet.getPluginsDir(). Lets refer this directory as PluginsSockDir.
+For any discovered plugin, pluginwatcher issues Registration.GetInfo grpc call
+to get plugin type, name and supported service API versions. For any registered plugin type,
+pluginwatcher calls the registered callback function with the received plugin
+name, supported service API versions, and the full socket path. The Kubelet
+component that receives this callback can acknowledge or reject the plugin
+according to its own logic, and use the socket path to establish its service
+communication with any API version supported by the plugin.
+
+Here are the general rules that Kubelet plugin developers should follow:
+- Run as priviledged. Currently creating socket under PluginsSockDir requires
+  priviledged access.
```

Comment:
Further down, someone mentions namespacing the plugin names.. another place namespacing may make sense is under the `PluginsSockDir`.  Maybe separate plugins socket path based on type i.e. `$PluginsSockDir/csi/<plugins-name>` or `$PluginsSockDir/devices/<plugins-name>` etc.